src/util/SharedMemory.cpp

```diff
@@ -129,7 +138,29 @@ bool SharedMemory::EnsureCommitted(int size)
     }
     return true;
 #else
-    return false;
+    if( size > max_ )
+        return false;
+    if( size > current_ )
+    {
+        unsigned new_size = current_;
+        while( size > new_size )
+            new_size += windowSize_;
+        void* old_handle = regionHandle;
+        void* new_handle = malloc(new_size);
+        if( new_handle && old_handle )
+        {
+            memcpy(new_handle,old_handle,current_);
+            free(old_handle);
+        }
+        if( new_handle )
+        {
+            current_ = new_size;
+            regionHandle = new_handle;
+            return true;
+        }else
+            return false;
+    }else
+        return true;
```

Comment:
just have a single default `return true` at the end and swap to `if (!handle) return false;`